src/pages/settings/Profile/PersonalDetails/AddressPage.js

```diff
@@ -59,10 +67,12 @@ function updateAddress(values) {
     PersonalDetails.updateAddress(values.addressLine1.trim(), values.addressLine2.trim(), values.city.trim(), values.state.trim(), values.zipPostCode.trim().toUpperCase(), values.country);
 }
 
-function AddressPage({privatePersonalDetails}) {
+function AddressPage({privatePersonalDetails, route}) {
     usePrivatePersonalDetails();
     const {translate} = useLocalize();
-    const [currentCountry, setCurrentCountry] = useState(PersonalDetails.getCountryISO(lodashGet(privatePersonalDetails, 'address.country')));
+    const countryFromUrl = lodashGet(route, 'params.country');
+    const [currentCountry, setCurrentCountry] = useState(countryFromUrl || PersonalDetails.getCountryISO(lodashGet(privatePersonalDetails, 'address.country')));
+    const [didCountryChange, setDidCountryChange] = useState(false);
```

Comment:
@parasharrajat So this doesn't cause infinite rerender? Did you put a `console.log `in the `useEffect `and try changing the country?
```
useEffect(() => {
      // This will cause the form to revalidate and remove any error related to country name
      onInputChange(countryCode);
  }, [countryCode]);
```

Reply:
Nope. Pushed the changes